src/math/p5.Vector.js

```diff
@@ -118,23 +127,225 @@ function vector(p5, fn) {
     // This is how it comes in with createVector()
     // This check if the first argument is a function
     constructor(...args) {
-      let x, y, z;
+      let dimensions = args.length; // TODO: make default 3 if no arguments
+      let values = args.map(arg => arg || 0);
       if (typeof args[0] === 'function') {
         this.isPInst = true;
         this._fromRadians = args[0];
         this._toRadians = args[1];
-        x = args[2] || 0;
-        y = args[3] || 0;
-        z = args[4] || 0;
-        // This is what we'll get with new p5.Vector()
+        values = args.slice(2).map(arg => arg || 0);
+      }
+      if (dimensions === 0) {
+        this.dimensions = 2;
+        this._values = [0, 0, 0];
```

Comment:
Should the dimensions here match the number of items in `_values`?

Reply:
One thing I'm just now thinking about, should the default vector when no argument is passed be 2D instead of 3D? Seeing as most people start learning to create sketches with 2D before moving into 3D.

Reply:
I think that makes sense. Even in webgl mode, you may very well still be using 2D vectors for a number of things, so its not a bad idea to have to be explicit to make 3D vectors